File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/Gethostbyname.java
package com.onevgo.functions;

import java.net.InetAddress;
import java.net.UnknownHostException;

public class Gethostbyname {
    public static String gethostbyname(String hostname) {
        try {
            return InetAddress.getByName(hostname).getHostAddress();
        } catch (UnknownHostException e) {
            e.printStackTrace();
        }
        return hostname;
    }

    public static void main(String[] args) {
        System.out.println(gethostbyname("www.onevgo.com"));
        System.out.println(gethostbyname("ss.onevgo.com"));
    }
}

File: 01_Language/01_Functions/go/src/functions/gethostbyname.go
package functions

import (
	"log"
	"net"
)

func Gethostbyname(hostname string) string {
	addrs, err := net.LookupHost(hostname)
	if err != nil {
		log.Println(err)
		return hostname
	}
	return addrs[0]
}
